React/wbooks/src/app/components/RouteHandler/index.js

```diff
@@ -11,27 +11,23 @@ class RouteHandler extends Component {
 
   render() {
     const { component: Element, ...rest } = this.props;
-    return (
-      <Route
-        {...rest}
-        render={props =>
-          this.props.isLogged ? (
-            this.isPublicRequest(props) ? (
-              <Redirect to="/dashboard" />
-            ) : (
-              <div>
-                <Navbar />
-                <Element {...props} />
-              </div>
-            )
-          ) : !this.isPublicRequest(props) ? (
-            <Redirect to="/" />
-          ) : (
+    const render = props =>
```

Comment:
Not here! not in render, you're declaring this function in each render! move as a class instance function!